client/blocks/login/index.jsx

```diff
@@ -79,6 +80,10 @@ class Login extends Component {
 		// Redirects to / if no redirect url is available
 		const url = redirectTo ? redirectTo : window.location.origin;
 
+		// user data is persisted in localstorage at `lib/user/user` line 157
+		// therefor we need to reset it before we redirect, otherwise we'll get
+		// mixed data from old and new user
+		store.clear();
```

Comment:
Instead of clearing all of local storage, which could contain information we want to keep (`debug`, a/b test perhaps), I think we should just clear the user data like we do [when logging out from `/me`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/user/utils.js#L52).